### PR TITLE
subsection tree and modularization

### PR DESCRIPTION
**Pull request by jkao1**
Depth is not infinite; currently the max is two. However, that just requires a few more while loops, which I will do in another PR. This needs to be merged into develop so @JasonLin43212 can complete his Footer.

### CODE REVIEW

**Comment by NicholasLYang** [comment]
I would consider turning the section routes and article routes into a separate component. Also, do not initialize subsection as null. Initialize as an empty object

**Comment by jkao1** [comment]
what's the purpose of making route components?

**Comment by NicholasLYang** [comment]
There's a lot of logic in the RoutingApp. Not sure it belongs there. It'd make more sense in a SectionRoutes component. Also, then you could connect that to the store instead of manually using the store

**Comment by NicholasLYang** [comment]
What's with the JSS naming? `Figure__image`?

**Comment by jkao1** [comment]
that's css convention. i wont budge on that one.

**Comment by NicholasLYang** [comment]
CSS convention != JSS convention

File: src/js/modules/sections/selectors.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+export const getAllSectionRoutes = createSelector(
+  getSections,
+  (sections) => {
+    let sectionRoutes = {};
+    Object.keys(sections).map(function (key) {
+      const section = sections[ key ];
+      let pathToSectionPage = '/' + section.slug;
+      let subsections = {};
+      if (section.parentSlug !== null) { // this section is a subsection
+        pathToSectionPage = '/' + section.parentSlug + '/' + section.slug;
```
Comment (NicholasLYang) [inline src/js/modules/sections/selectors.js]: Minor ass thing, but try to use string interpolation instead of concatenation

**Comment by NicholasLYang** [comment]
We gotta go over this a little more carefully

**Pull request merged**